### DISCUSSION TITLE
بررسی وضعیت تراکنش قبل از تایید

### DISCUSSION THREAD

**Discussion by masoud86** [Sina-Soltani/Parbad#186] [Q&A]
سلام
یه مشکلی در بررسی وضعیت پرداخت بعد از انتقال از بانک برخوردم.
بعد از برگشت از درگاه بانک و قبل از تایید پرداخت با استفاده از متد
Verify
باید شرایط مجدد چک بشه که طبق این مثال پیاده سازی کردم.
```C#
if (!Is_There_Still_Product_In_Shop(invoice.TrackingNumber))
            {
                var cancelResult = await _onlinePayment.CancelAsync(invoice, cancellationReason: "Sorry, We have no more products to sell.");

                return View("CancelResult", cancelResult);
            }

            var verifyResult = await _onlinePayment.VerifyAsync(invoice);
```
مشکل جایی پیش میاد که کاربر پرداخت رو داخل درگاه کنسل میکنه و تا قبل از استفاده از
Verify
راهی برای متوجه شدن ندارم. یعنی در صورتیکه کاربر کنسل کرده باشه باز هم مجبورم که شرایط رو چک کنم و پرداخت رو کنسل کنم در صورتیکه اصلاً پرداختی انجام نشده.

**Comment by Sina-Soltani** [comment]
درود

درسته. این قابلیت باید اضافه بشه

میشه لطفا یه تیکت ایجاد کنید برای این مشکل؟
[New Issue](https://github.com/Sina-Soltani/Parbad/issues/new?assignees=Sina-Soltani&labels=feature&template=requesting-a-new-feature.md&title=)

**Reply by masoud86** [reply]
ممنون از پاسخ سریع شما
تیکت ایجاد شد
